frontend/public/components/events.tsx

```diff
@@ -0,0 +1,267 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import classNames from 'classnames';
+import { Link } from 'react-router-dom';
+import { Helmet } from 'react-helmet';
+
+import { ResourceListDropdown } from './resource-dropdown';
+import { TextFilter } from './factory';
+import { referenceFor, K8sResourceKind, EventInvolvedObject, EventKind } from '../module/k8s';
+import { withStartGuide } from './start-guide';
+import { NodeModel } from '../models';
+import { connectToFlags } from '../reducers/features';
+import { FLAGS } from '../const';
+import {
+  Dropdown,
+  PageHeading,
+  ResourceLink,
+  resourcePathFromModel,
+  Timestamp,
+  TogglePlay,
+  Box,
+  Loading,
+  pluralize,
+} from './utils';
+import { EventStream, EventStreamList, maxMessages, EventStreamChildProps } from './utils/events-stream';
```

Comment:
Can you describe the effective changes between original `events.jsx` and new `events.tsx`? (I wish the former was git-move'd to the latter for better diff output..)

Is it just about moving event handling logic into separate module (event stream) and adding type annotations?

Reply:
Previously `EventStream` was a big component which was responsible for fetching and rendering events. In order to reuse the fetching logic for Dashboard Events card, I had to decouple it. I split it into smaller components. 
- `EventStream` is now only fetching the events and injects them as props into its children
- `EventStreamList` renders events passed via props into Virtualized List.
I put those components into `utils/event-stream` as `EventStream` is similar to `Firehose` (fetches something and passes the result to children)

With the given change I had to adjust components in `events.jsx` and during the process I also added types.